File: src/java/magma/app/compile/type/Variadics.java
package magma.app.compile.type;

import magma.app.compile.rule.NodeRule;
import magma.app.compile.rule.Rule;
import magma.app.compile.rule.StripRule;
import magma.app.compile.rule.SuffixRule;

public class Variadics {
    public static StripRule createVariadicRule(Rule typeRule) {
        return new StripRule(new SuffixRule("...", new NodeRule("child", typeRule)));
    }
}

File: src/ts/magma/app/compile/type/Variadics.ts
import { Node } from "../../../../magma/app/compile/node/Node";
import { Rule } from "../../../../magma/app/compile/rule/Rule";
import { TypeRule } from "../../../../magma/app/compile/rule/TypeRule";
import { StripRule } from "../../../../magma/app/compile/rule/StripRule";
import { SuffixRule } from "../../../../magma/app/compile/rule/SuffixRule";
import { NodeRule } from "../../../../magma/app/compile/rule/NodeRule";
import { TypeCompiler } from "../../../../magma/app/TypeCompiler";
export class Variadics {
	static createVariadicRule(): Rule<Node> {
		return new TypeRule("variadic", new StripRule(new SuffixRule<Node>("...", new NodeRule("child", TypeCompiler.lexType))))/*unknown*/;
	}
}
